### PR TITLE
fix(client): Ensure temporary migration folder deletion

### PR DESCRIPTION
**Pull request by msfstef**
When client generation failed, the process exited before having a chance to delete the temporary migrations folder, leaving you with an ever increasing number of failed temporary migration folders.

They were also tracked on git by default so I added them to the starter template gitignore - not the most maintainable solution but helps keep things neat for the starter experience.

### CODE REVIEW

**Review by kevin-dp — COMMENTED** [review]
Implementation looks good!
Though i'm not sure we want to delete the folder if something goes wrong.
Often times this folder is very useful to debug what went wrong as it gives us access to the intermediate Prisma schema for which client generation fails. Perhaps also make that configurable? e.g. by default we always remove the temporary directory but if you enable the flag we don't delete it on error.

**Comment by kevin-dp** [comment]
Btw, you should generate a changeset for each PR, to do so run `pnpm changeset` at the root of the mono repo and follow the steps of the interactive changeset tool.

**Comment by msfstef** [comment]
Added a generic ```--debug``` flag to the generator which can be reused for maintaining more info - figured I'd move around some things to using switch statements for clarity.

Added a changeset, tried to include all changes in it so it's multiline, hope that's ok!

File: clients/typescript/src/cli/migrations/migrate.ts [hunk 1]
```diff
@@ -211,11 +214,20 @@ async function _generate(opts: Omit<GeneratorOptions, 'watch'>) {
     await buildMigrations(migrationsFolder, migrationsFile)
     console.log('Successfully built migrations')
   } catch (e: any) {
+    generationFailed = true
     console.error('generate command failed: ' + e)
-    process.exit(1)
+
+    // Rethrow error - finally statement still runs
+    if (!opts.exitOnError) throw e
```
Comment (kevin-dp) [inline clients/typescript/src/cli/migrations/migrate.ts:221]: Why is this if test needed? If `opts.exitOnError` is `true` and we would still `throw e` then we would still get into the `finally` block and we would call `process.exit(1)` on L230.

File: clients/typescript/test/cli/migrations/migrate.test.ts [hunk 2]
```diff
@@ -108,9 +113,71 @@ model Model {
...
+    // should fail generaton
+    t.assert(await failedGenerate(false))
+
+    // should clean up temporary folders
+    t.assert((await findMigrationFolder()) === null)
+  }
+)
+
+test.serial(
+  'migrator should retain temporary folder on falure in debug mode',
```
Comment (kevin-dp) [inline clients/typescript/test/cli/migrations/migrate.test.ts:171]: typo in this test name `falure` -> `failure`

**Review by kevin-dp — APPROVED** [review]
Your changes look good!
Left 2 minor comments but when can merge after that.

**Comment by kevin-dp** [comment]
I see that you merged main into your branch to fix some conflicts.
We usually don't do that but solve conflicts by rebasing on top of main to keep the history linear.

File: clients/typescript/src/cli/migrations/migrate.ts (hunk 1 above)
Comment (msfstef) [reply clients/typescript/src/cli/migrations/migrate.ts:221]: good point - definitely not needed!

**Comment by samwillis** [comment]
@msfstef I would hold off on modifying the commands too much as #664 refactors them a lot and we will need to add any new arguments there.

**Pull request merged**